client/core/bookie.go

```diff
@@ -395,6 +398,210 @@ func handleBookOrderMsg(_ *Core, dc *dexConnection, msg *msgjson.Message) error
+	}
+
+	err = book.Reset(&msgjson.OrderBook{
+		MarketID: sp.MarketID,
+		Seq:      sp.Seq,        // forces seq reset, but should be in seq with previous
+		Epoch:    sp.FinalEpoch, // unused?
+		// Orders is nil
+	})
+	// Return any non-nil error, but still revoke purged orders.
+
+	// Revoke all active orders of the suspended market for the dex.
+	log.Warnf("Revoking all active orders for market %v.", mkt.Name)
+	updatedAssets := make(assetMap)
+	dc.tradeMtx.RLock()
+	for _, tracker := range dc.trades {
+		if tracker.Order.Base() == mkt.BaseID && tracker.Order.Quote() == mkt.QuoteID &&
+			tracker.metaData.Host == dc.acct.host && tracker.metaData.Status == order.OrderStatusBooked {
+			// Locally revoke the purged book order.
+			tracker.revoke()
+			updatedAssets.count(tracker.fromAssetID)
+		}
+	}
+	dc.tradeMtx.RUnlock()
+
+	// Clear the book.
+	book.send(&BookUpdate{
+		Action:   FreshBookAction,
+		Host:     dc.acct.host,
+		MarketID: sp.MarketID,
+		Payload: &MarketOrderBook{
+			Base:  mkt.BaseID,
+			Quote: mkt.QuoteID,
+			Book:  book.book(), // empty
+		},
+	})
+
+	dc.refreshMarkets()
+	c.updateBalances(updatedAssets)
+
+	return err
+}
+
+// handleTradeResumptionMsg is called when a trade resumption notification is
+// received. This may be an orderbook message at the time of resumption or a
+// general notification at any time prior to the market suspend.
+func handleTradeResumptionMsg(c *Core, dc *dexConnection, msg *msgjson.Message) error {
+	var rs msgjson.TradeResumption
+	err := msg.Unmarshal(&rs)
+	if err != nil {
+		return fmt.Errorf("trade resumption unmarshal error: %v", err)
+	}
+
+	// Ensure the provided market exists for the dex.
+	if dc.market(rs.MarketID) == nil {
+		return fmt.Errorf("no market found with ID %s", rs.MarketID)
+	}
+
+	// rs.ResumeTime == 0 means resume now.
+	if rs.ResumeTime != 0 {
+		// This is just a notice about a scheduled resumption.
+		dc.setMarketStartEpoch(rs.MarketID, rs.StartEpoch, false) // set the start epoch, leaving any final/persist data
+		resTime := encode.UnixTimeMilli(int64(rs.ResumeTime))
+		detail := fmt.Sprintf("Market %s is now scheduled for resumption at %v", rs.MarketID, resTime)
+		c.notify(newServerNotifyNote("market resume scheduled", detail, db.WarningLevel))
+		return nil
+	}
+
+	// Update the market's status and mark the new epoch.
+	dc.setMarketStartEpoch(rs.MarketID, rs.StartEpoch, true) // and clear the final/persist data
+	dc.epochMtx.Lock()
+	dc.epoch[rs.MarketID] = rs.StartEpoch
+	dc.epochMtx.Unlock()
+
+	// TODO: Server config change without restart is not implemented on the
+	// server, but it would involve either getting the config response or adding
+	// the entire market config to the TradeResumption payload.
+	//
+	// Fetch the updated DEX configuration.
+	// dc.fetchServerConfig()
+
+	detail := fmt.Sprintf("Market %s has resumed trading at epoch %d",
+		rs.MarketID, rs.StartEpoch)
+	c.notify(newServerNotifyNote("market resumed", detail, db.WarningLevel))
```

Comment:
How about `db.SuccessLevel`?

Reply:
Funny, I started with that and felt like something a little louder was needed.  Although I had reservations about the yellow dot instead of green since that implies something is not quite right.  Will go back to SuccessLevel, but I think we might consider something new that has perhaps a blue dot.